BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -552,62 +596,77 @@ - (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticatio
                                                                                                  message:nil
                                                                                           preferredStyle:UIAlertControllerStyleAlert];
-                        
+
                         UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"Cancel", @"Label: Cancel action. Usually used on buttons.")
-                                                                           style:UIAlertActionStyleCancel
-                                                                         handler:^(UIAlertAction *action) {
-                                                                             [alertController dismissViewControllerAnimated:YES completion:nil];
-                                                                             BOXLog(@"Cancel");
-                                                                         }];
+                                                                               style:UIAlertActionStyleCancel
+                                                                             handler:^(UIAlertAction *action) {
+                                                                                 BOXLog(@"Cancel");
+                                                                                 if (completionHandler) {
+                                                                                     completionHandler(NSURLSessionAuthChallengeCancelAuthenticationChallenge, NULL);
+                                                                                 } else {
+                                                                                     [[self.authenticationChallenge sender] cancelAuthenticationChallenge:self.authenticationChallenge];
+                                                                                 }
+                                                                                 [alertController dismissViewControllerAnimated:YES completion:nil];
+                                                                             }];
                         UIAlertAction *submitAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"Submit", @"Alert view button: submit button for authentication challenge")
                                                                                style:UIAlertActionStyleDefault
                                                                              handler:^(UIAlertAction *action) {
                                                                                  [alertController dismissViewControllerAnimated:YES completion:nil];
-                                                                                 
+
```

Comment:
minor: not sure if calling this too early could prevent the rest from execution, or should we call [alertController dismissViewControllerAnimated:YES completion:nil]; at the end of the block.